mobgap/gsd/evaluation.py

```diff
@@ -402,59 +417,165 @@ def _get_false_matches_from_overlap_data(overlaps: list[Interval], interval: Int
+        handle.set_linewidth(10)
+    plt.tight_layout()
+    return fig
+
+
+def _sanitize_index(ic_list: pd.DataFrame, list_type: Literal["detected", "reference"]) -> tuple[pd.DataFrame, bool]:
+    is_multindex = False
+    # check if index is a multiindex and raise warning if it is
+    if isinstance(ic_list.index, pd.MultiIndex):
+        is_multindex = True
+        ic_list.index = ic_list.index.to_flat_index()
+        ic_list.index.name = f"gs_id_{list_type}"
+    # check if indices are unique
+    if not ic_list.index.is_unique:
+        raise ValueError(f"The index of `gs_list_{list_type}` must be unique!")
+    return ic_list, is_multindex
+
+
+def _plot_intervals_from_df(df: pd.DataFrame, y: int, ax: Axes, **kwargs: Unpack[dict[str, Any]]) -> None:
+    label_set = False
+    for _, row in df.iterrows():
+        label = kwargs.pop("label", None)
+        if label and not label_set:
+            ax.hlines(y, row["start"], row["end"], lw=20, label=label, **kwargs)
+            label_set = True
+        else:
+            ax.hlines(y, row["start"], row["end"], lw=20, **kwargs)
+
+
+def categorize_matches_with_min_overlap(
+    *,
+    gsd_list_detected: pd.DataFrame,
+    gsd_list_reference: pd.DataFrame,
+    overlap_threshold: float = 0.8,
+    multiindex_warning: bool = True,
+) -> pd.DataFrame:
+    """Evaluate a gait sequence list against a reference sequence-by-sequence with a minimum overlap threshold.
+
+    This compares a gait sequence list against a reference list and classifies each detected sequence as true positive,
+    false positive, or false negative.
+    A gait sequence is classified as true positive when having at least ``overlap_threshold`` overlap with a reference
+    sequence. If a detected sequence has no overlap with any reference sequence, it is classified as false positive.
+    If a reference sequence has no overlap with any detected sequence, it is classified as false negative.
 
     Note, that the threshold is enforced in both directions. That means, that the relative overlap of the detected gait
     sequence with respect to the overall length of the detected interval AND to the overall length of the matched
     reference interval must be at least `overlap_threshold`.
 
+    The detected and reference dataframes are expected to have columns namend "start" and "end" containing the
+    start and end indices of the respective gait sequences.
+    As index, we support either a single or a multiindex without duplicates
+    (i.e., the index must identify each gait sequence uniquely).
+    If a multiindex is provided, the single index levels will be ignored for the comparison and matches across different
+    index groups will be possible.
+    If this is not the intended use case, consider grouping your input data before calling this function
+    (see :func:`~mobgap.utils.array_handling.create_multi_groupby`).
+
     Note, we assume that ``gsd_list_detected`` has no overlaps, but we don't enforce it!
```

Comment:
Add a clearly visible note on the differences between this and the `categorize_interval` method. Namely, that this will not create new intervals, but only return values from the input.